packages/fx-core/src/component/driver/teamsApp/utils/ManifestUtils.ts

```diff
@@ -393,6 +391,32 @@ export class ManifestUtils {
     const manifest = JSON.parse(manifestString) as TeamsAppManifest;
     return ok(manifest);
   }
+
+  /**
+   * trim the short name in manifest to make sure it is no more than 25 length
+   */
+  async trimManifestShortName(
+    projectPath: string,
+    maxLength = 25
+  ): Promise<Result<undefined, FxError>> {
+    const manifestPath = this.getTeamsAppManifestPath(projectPath);
+    const manifest = (await fs.readJson(manifestPath)) as TeamsAppManifest;
+    const shortName = manifest.name.short;
+    let hasSuffix = false;
+    let trimmedName = shortName;
+    if (shortName.includes("${{APP_NAME_SUFFIX}}")) {
+      hasSuffix = true;
+      trimmedName = shortName.replace("${{APP_NAME_SUFFIX}}", "");
+    }
+    if (trimmedName.length <= maxLength) return ok(undefined);
+    let newShortName = trimmedName.replace(/\s/g, "").slice(0, maxLength);
+    if (hasSuffix) {
+      newShortName += "${{APP_NAME_SUFFIX}}";
```

Comment:
`newShortName` can exceed 25 length since you add a suffix placeholder?

Reply:
Max length of suffix is 'local', so max length for the name is 25.